### ISSUE TITLE
Updating Next.js from `12.0.0` to `12.0.5` breaks `Document` type

### ISSUE THREAD

**Issue by williamhzo** [vercel/next.js#32563]
### What version of Next.js are you using?

12.0.5

### What version of Node.js are you using?

16.13.0

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying your application?

Nx

### Describe the Bug

Bumping Next.js from `12.0.0` -> `12.0.5` (due the [security alert](https://github.com/vercel/next.js/security/advisories/GHSA-25mp-g6fv-mqxx)) breaks `lint` and `build` due to a type error, as seen in screenshots below:

<img width="688" alt="Screenshot 2021-12-16 at 11 34 39" src="https://user-images.githubusercontent.com/49357401/146355821-6b874164-b827-460d-af4a-2124ac9dadc2.png">

### Expected Behavior

This minor bump (`12.0.0` -> `12.0.5`) should not break `lint` and `build`, no error should be thrown here.

### To Reproduce

[Custom Next.js document](https://nextjs.org/docs/advanced-features/custom-document#customizing-renderpage) `_document.tsx` file with `styled-components` config, in Next.js `12.0.0`:

```ts
import Document from 'next/document'

class MyDocument extends Document {
  static async getInitialProps(ctx) {
    const originalRenderPage = ctx.renderPage

    ctx.renderPage = () =>
      originalRenderPage({
        // useful for wrapping the whole react tree
        enhanceApp: (App) => App,
        // useful for wrapping in a per-page basis
        enhanceComponent: (Component) => Component,
      })

    // Run the parent `getInitialProps`, it now includes the custom `renderPage`
    const initialProps = await Document.getInitialProps(ctx)

    return initialProps
  }
}

export default MyDocument
```

The error occurs when bumping to `12.0.5`.

**Comment by awareness481** [comment]
It's fixed, upgrade to `12.0.6` or newer

**Comment by williamhzo** [comment]
> It's fixed, upgrade to `12.0.6` or newer

great, thanks! closing the issue.

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
